draco/tests/test_hard.py

```diff
@@ -1381,3 +1381,262 @@ def test_aggregate_not_all_continuous():
         )
         == ["aggregate_not_all_continuous"]
     )
+
+
+def test_invalid_bin():
+    b = hard.blocks["invalid_bin"]
+    assert isinstance(b, Block)
+    p = b.program
+
+    assert no_violations(
+        p
+        + """
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+    attribute((encoding,field),2,temperature).
+    attribute((encoding,binning),2,30).
+    """
+    )
+
+    assert no_violations(
+        p
+        + """
+    entity(encoding,1,2).
+    attribute((encoding,channel),2,x).
+    attribute((encoding,field),2,temperature).
+    attribute((encoding,binning),2).
+    """
+    )
+
+    assert (
```

Comment:
Add a comment explaining the issue in each of the examples below.

Reply:
added comments